pkg/fleet/installer/setup/djm/emr.go

```diff
@@ -0,0 +1,181 @@
+
+	s.Config.DatadogYAML.ExpectedTagsDuration = "10m"
+	isMaster, clusterName, err := setupCommonEmrHostTags(s)
+	if err != nil {
+		return fmt.Errorf("failed to set tags: %w", err)
+	}
+	if isMaster {
+		setupEmrDriver(s, clusterName)
+	}
+	return nil
+}
+
+func setupCommonEmrHostTags(s *common.Setup) (bool, string, error) {
+
+	instanceInfoRaw, err := os.ReadFile(filepath.Join(emrInfoPath, "instance.json"))
+	if err != nil {
+		return false, "", fmt.Errorf("error reading instance file: %w", err)
+	}
+
+	var info emrInstanceInfo
+	if err = json.Unmarshal(instanceInfoRaw, &info); err != nil {
+		return false, "", fmt.Errorf("error umarshalling instance file: %w", err)
+	}
+
+	setHostTag(s, "instance_group_id", info.InstanceGroupID)
+	setHostTag(s, "is_master_node", strconv.FormatBool(info.IsMaster))
+	s.Span.SetTag("host_tag."+"is_master_node", info.IsMaster)
+
+	extraInstanceInfoRaw, err := os.ReadFile(filepath.Join(emrInfoPath, "extraInstanceData.json"))
+	if err != nil {
+		return info.IsMaster, "", fmt.Errorf("error reading extra instance data file: %w", err)
+	}
+
+	var extraInfo extraEmrInstanceInfo
+	if err = json.Unmarshal(extraInstanceInfoRaw, &extraInfo); err != nil {
+		return info.IsMaster, "", fmt.Errorf("error umarshalling extra instance data file: %w", err)
+	}
+	setHostTag(s, "job_flow_id", extraInfo.JobFlowID)
+	setHostTag(s, "cluster_id", extraInfo.JobFlowID)
+	s.Span.SetTag("emr_version", extraInfo.ReleaseLabel)
+
+	emrResponseRaw, err := executeCommandWithTimeout("aws", "emr", "describe-cluster", "--cluster-id", extraInfo.JobFlowID)
+	if err != nil {
+		return info.IsMaster, extraInfo.JobFlowID, fmt.Errorf("error describing emr cluster, using cluster id as name: %w", err)
+	}
+	var response emrResponse
+	if err = json.Unmarshal(emrResponseRaw, &response); err != nil {
+		return info.IsMaster, extraInfo.JobFlowID, fmt.Errorf("error unmarshalling AWS EMR response,  using cluster id as name: %w", err)
+	}
+
+	setHostTag(s, "cluster_name", response.Cluster.Name)
+	return info.IsMaster, response.Cluster.Name, nil
+}
+
+func setupEmrDriver(s *common.Setup, clusterName string) {
+
+	s.Config.InjectTracerYAML.AdditionalEnvironmentVariables = tracerEnvConfigEmr
+
+	var sparkIntegration common.IntegrationConfig
+	var yarnIntegration common.IntegrationConfig
+
+	if clusterName != "" {
+		sparkIntegration.Instances = []any{
+			common.IntegrationConfigInstanceSpark{
+				SparkURL:         "http://127.0.0.1:8088",
+				SparkClusterMode: "spark_yarn_mode",
+				ClusterName:      clusterName,
+				StreamingMetrics: false,
+			},
+		}
+		yarnIntegration.Instances = []any{
+			common.IntegrationConfigInstanceYarn{
+				ResourceManagerURI: "http://127.0.0.1:8088",
+				ClusterName:        clusterName,
+			},
+		}
+		s.Config.IntegrationConfigs["spark.d/conf.yaml"] = sparkIntegration
+		s.Config.IntegrationConfigs["yarn.d/conf.yaml"] = yarnIntegration
+	} else {
+		log.Warn("clusterName is empty, Spark and yarn integrations are not set up")
```

Comment:
nit, the usual go pattern is to exit early, it's easier to read

if clusterName == "" {
     log.Warn...
     return
 }
// golden path with no indent